File: source/rolx-client/src/app/account/core/project.ts
import { Type } from 'class-transformer';
import moment from 'moment';
import { Phase } from './phase';

export class Project {

  id: number;
  number: string;
  name: string;

  @Type(() => Phase)
  phases: Phase[] = [];

  get nextNumber() {
    return Math.max(...this.phases.map(ph => ph.number), 0) + 1;
  }

  addPhase(): Phase {
    const phase = new Phase();
    phase.startDate = moment();
    phase.number = this.nextNumber;
    this.phases.push(phase);
    return phase;
  }
}

File: source/RolXServer/RolXServer/Projects/WebApi/Resource/Project.cs
// -----------------------------------------------------------------------
// <copyright file="Project.cs" company="Christian Ewald">
// Copyright (c) Christian Ewald. All rights reserved.
// Licensed under the MIT license.
// See LICENSE.md in the project root for full license information.
// </copyright>
// -----------------------------------------------------------------------

using System.Collections.Generic;

namespace RolXServer.Projects.WebApi.Resource
{
    /// <summary>
    /// A project we are working on.
    /// </summary>
    public class Project
    {
        /// <summary>
        /// Gets or sets the identifier.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Gets or sets the number.
        /// </summary>
        public string Number { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the name.
        /// </summary>
        public string Name { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the phases.
        /// </summary>
        public List<Phase> Phases { get; set; } = new List<Phase>();
    }
}
